v3/how-to-guides/02-pallet-design/a-add-contracts-pallet/index.mdx

```diff
@@ -10,34 +10,58 @@ section: how to guides
 category: pallet design
 ---
 
-This guide will show you how to add the [Contracts pallet](/rustdocs/latest/pallet_contracts/index.html) to your runtime so that your blockchain can support Wasm smart contracts.
+This guide will show you how to add the [Contracts pallet](/rustdocs/latest/pallet_contracts/index.html) to your runtime so that your blockchain can support WebAssembly (Wasm) smart contracts.
+
 You can follow similar patterns to add additional FRAME pallets to your runtime, however you should note that each pallet is a little different in terms of the specific configuration settings needed to use it correctly.
 
+## What is the Contracts Pallet for?
+
+Broadly speaking there are two big use-cases you can apply it to.
+We'll only shortly describe those, for more information please check
+[the pallet reamde][pallet-readme].
+
+### Smart contracts as first-class citizens
+
+You want to build a general-purpose smart contracts chain, with them being the main focus of your chain.
+Maybe you have an innovative idea and want to have a smart contracts base layer which you complement
+with your chains unique selling point.
+
+### Smart contracts for users to customize how they interact with your chain
+
+You've built a chain that doesn't have smart contracts as first-class citizens,
+instead most of the logic is built using Substrate pallets that use weight-metering.
```

Comment:
Weight-metering isn't a thing. Metering is understood by most people as "high overhead" technique what pallets are avoiding.

```suggestion
instead most of the logic is built using Substrate pallets.
```